src/main/java/io/knotx/commons/cache/inmemory/InMemoryCacheOptions.java

```diff
@@ -0,0 +1,204 @@
+ */
+package io.knotx.commons.cache.inmemory;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+import java.util.Objects;
+
+@DataObject(generateConverter = true)
+public class InMemoryCacheOptions {
+
+  private boolean enableMaximumSize = true;
+  private boolean enableTtlAfterWrite = true;
+  private boolean enableTtlAfterAccess = false;
+  private long ttl = 5000L;
+  private long ttlAfterAccessMs = 5000L;
+  private Long ttlAfterWriteMs;
+  private long maximumSize = 1000L;
+
+  public InMemoryCacheOptions(JsonObject json) {
+    InMemoryCacheOptionsConverter.fromJson(json, this);
+  }
+
+  public JsonObject toJson() {
+    JsonObject output = new JsonObject();
+    InMemoryCacheOptionsConverter.toJson(this, output);
+    return output;
+  }
+
+  public boolean isEnableMaximumSize() {
+    return enableMaximumSize;
+  }
+
+  /**
+   * Sets flag controlling whether cache size should be limited by number of entries. When enabled,
+   * maximumSize option is passed to Cache builder.
+   * <p>
+   * Defaults to true.
+   *
+   * @param enableMaximumSize should size of cache be limited by number of entries
+   * @return a reference to this so that API can be used fluently
+   */
+  public InMemoryCacheOptions setEnableMaximumSize(boolean enableMaximumSize) {
+    this.enableMaximumSize = enableMaximumSize;
+    return this;
+  }
+
+  public boolean isEnableTtlAfterWrite() {
+    return enableTtlAfterWrite;
+  }
+
+  /**
+   * Sets flag controlling whether TTL after write should be configured. When enabled, ttlAfterWrite
+   * or ttl option is passed to Cache builder.
+   * <p>
+   * Defaults to true.
+   *
+   * @param enableTtlAfterWrite should TTL after write be configured
+   * @return a reference to this so that API can be used fluently
+   */
+  public InMemoryCacheOptions setEnableTtlAfterWrite(boolean enableTtlAfterWrite) {
+    this.enableTtlAfterWrite = enableTtlAfterWrite;
+    return this;
+  }
+
+  public boolean isEnableTtlAfterAccess() {
+    return enableTtlAfterAccess;
+  }
+
+  /**
+   * Sets flag controlling whether TTL after access should be configured. When enabled, ttlAfterAccess
+   * option is passed to Cache builder.
+   * <p>
+   * Defaults to false.
+   *
+   * @param enableTtlAfterAccess should TTL after access be configured
+   * @return a reference to this so that API can be used fluently
+   */
+  public InMemoryCacheOptions setEnableTtlAfterAccess(boolean enableTtlAfterAccess) {
+    this.enableTtlAfterAccess = enableTtlAfterAccess;
+    return this;
+  }
+
+  public long getTtl() {
+    return ttl;
+  }
+
+  /**
+   * Legacy option for setting TTL after write in ms. Applies only when ttlAfterWrite is not set and
+   * enableTtlAfterWrite is set to true.
+   * <p>
+   * Defaults to 5000ms.
+   *
+   * @param ttl TTL after write in millis
+   * @return a reference to this so that API can be used fluently
+   */
+  @Deprecated
+  public InMemoryCacheOptions setTtl(long ttl) {
```

Comment:
Why do we need this deprecated option from the start of a new class? Isn't just `ttlAfterWrite` enough?

Reply:
I believe this option comes from previous implementation for Actions.